### PR TITLE
merge v5.4 into v6.0

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs [hunk 1]
```diff
@@ -433,6 +434,8 @@ public class FtpSettings : BackupSettings
 
         public string CertificateAsBase64 { get; set; }
 
+        public string CertificateThumbprint { get; private set; }
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: remove this

File: src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs [hunk 2]
```diff
@@ -474,11 +477,26 @@ public override DynamicJsonValue ToJson()
...
+                    // ignore
+                }
+            }
+
             var djv = base.ToAuditJson();
             djv[nameof(Url)] = Url;
-            djv[nameof(Port)] = Port;
             djv[nameof(UserName)] = UserName;
-            djv[nameof(CertificateFileName)] = CertificateFileName;
+            djv[nameof(CertificateThumbprint)] = CertificateThumbprint;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: not needed

File: src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs [hunk 3]
```diff
@@ -113,7 +113,7 @@ public async Task CreateInternalAsync(BlittableJsonReaderObject bjro, Subscripti
                         break;
 
                     case Constants.Documents.SubscriptionChangeVectorSpecialStates.LastDocument:
-                        options.ChangeVector = Database.DocumentsStorage.GetLastDocumentChangeVector(context.Transaction.InnerTransaction, context, sub.Collection);
+                        options.ChangeVector = GetLastDocumentChangeVectorForSubscription(Database, context, sub);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: this was also fixed in 'Try' IIRC

File: src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: removed

File: src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: removed

File: src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs (hunk 3 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: indeed

**Pull request merged**